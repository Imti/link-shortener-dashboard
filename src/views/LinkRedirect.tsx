import { useEffect } from 'react'
import { Link } from '../utils'

export default function LinkRedirect(props: Link) {
  const { uri, id } = props

  useEffect(() => {
    const links = JSON.parse(window.localStorage.getItem('links')!)
    const index = links.findIndex((link: Link) => link.id === id)

    // increment count on the selected link
    const updatedLink = { ...links[index], count: links[index].count + 1 }

    // reconstruct the new links
    const newLinks = [
      ...links.slice(0, index),
      updatedLink,
      ...links.slice(index + 1),
    ]

    window.localStorage.setItem('links', JSON.stringify(newLinks))
  }, [id])

  // TODO Determine if we want this or want to use react router Redirect component
  useEffect(() => {
    window.location.href = uri
  }, [uri])

  return (
    <span className="block text-xl md:text-2xl font-semibold text-slate-300 mb-1">
      Redirecting...
    </span>
  )
}
