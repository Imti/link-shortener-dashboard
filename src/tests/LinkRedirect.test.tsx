import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import LinkRedirect from '../views/LinkRedirect'
import { mockLinkProps } from './fixtures'

describe('<LinkRedirect />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <LinkRedirect {...mockLinkProps} />
        </Router>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
