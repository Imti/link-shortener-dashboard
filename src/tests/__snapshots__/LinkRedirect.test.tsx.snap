// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkRedirect /> renders correctly 1`] = `
<span
  className="block text-xl md:text-2xl font-semibold text-slate-300 mb-1"
>
  Redirecting...
</span>
`;
